src/optimal_image/search.rs

```diff
@@ -0,0 +1,58 @@
+use dataclients::ImageData;
+use std::path::Path;
+use ImageConfig;
+use ImageFormat;
+use ImageScore;
+
+pub struct SearchOptions {
+    threshold: f64,
+    formats: Vec<ImageFormat>,
+}
+
+pub struct Search {
+    image_data: ImageData,
+    options: SearchOptions,
+}
+
+impl Search {
+    fn from_path(&self, path: &Path) -> &Search {
+        // TODO: initialize struct members based on path
+        self
+
+
+
+    }
+}
+
+pub fn find_optimal_config(
```

Comment:
Why not part of `Search` struct?

Reply:
for now this is a utility function to the binary search we need, in the struct will have something that uses it, but will have to do also some extra steps:
- generate the vector of configs based on options (and group them by image format)
- run in parallel a find_optimal_config for each of the required image formats
- decide what to return (kind of a promise, maybe store the future result in the struct and have a getter to retrieve it without rerunning the function, etc...)